File: Reddit/app/src/main/java/com/thirdmono/reddit/data/entity/Listing.java
package com.thirdmono.reddit.data.entity;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class Listing {

    @SerializedName("modhash")
    private String modhash;
    @SerializedName("children")
    private List<Thing> things = null;
    @SerializedName("after")
    private String after;
    @SerializedName("before")
    private Object before;

    public String getModhash() {
        return modhash;
    }

    public void setModhash(String modhash) {
        this.modhash = modhash;
    }

    public List<Thing> getThings() {
        return things;
    }

    public void setThings(List<Thing> things) {
        this.things = things;
    }

    public String getAfter() {
        return after;
    }

    public void setAfter(String after) {
        this.after = after;
    }

    public Object getBefore() {
        return before;
    }

    public void setBefore(Object before) {
        this.before = before;
    }

}

File: Reddit/app/src/main/java/com/thirdmono/reddit/data/entity/Listing.kt
package com.thirdmono.reddit.data.entity

import com.google.gson.annotations.SerializedName

data class Listing (
    @SerializedName("modhash")
    private val modhash: String? = null,
    @SerializedName("children")
    private val things: List<Thing>? = null,
    @SerializedName("after")
    private val after: String? = null,
    @SerializedName("before")
    private val before: Any? = null
)